webpack.common.js

```diff
@@ -1,30 +1,45 @@
 const { ESBuildPlugin, ESBuildMinifyPlugin } = require('esbuild-loader');
 const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 
-module.exports = {
-  resolve: {
-    alias: {
-      'video.js': 'video.js/core',
+module.exports = (env) => {
+  const config = {
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          loader: 'esbuild-loader',
+        }
+      ]
     },
-  },
-  module: {
-    rules: [
-      {
-        test: /\.js$/,
-        loader: 'esbuild-loader',
-      }
+    optimization: {
+      minimize: true,
+      minimizer: [new ESBuildMinifyPlugin({
+        target: 'es2015'
+      })],
+    },
+    plugins: [
+      new ESBuildPlugin(),
+      new MiniCssExtractPlugin({
+        filename: 'index.css',
+      }),
     ]
-  },
-  optimization: {
-    minimize: true,
-    minimizer: [new ESBuildMinifyPlugin({
-      target: 'es2015'
-    })],
-  },
-  plugins: [
-    new ESBuildPlugin(),
-    new MiniCssExtractPlugin({
-      filename: 'index.css',
-    }),
-  ]
+  };
+
+  if (!env.vhs) {
+    config.resolve = {
+      alias: {
+        'video.js': 'video.js/core',
+      }
+    };
```

Comment:
When using hls.js, the default playback engine, we want to alias `video.js` to `video.js/core` so that VHS isn't included.

Reply:
Ahhh, very clever. To be clear:

* in Hls.js build: `import video.js` will import core only
* in VHS build: `import video.js` will import video.js